File: esports-backend/src/main/java/com/esports/dto/UserProfileUpdateDTO.java
package com.esports.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Pattern;
import lombok.Data;
import org.hibernate.validator.constraints.URL;

@Data
public class UserProfileUpdateDTO {
    @Pattern(regexp = "^[a-zA-Z]+(\s[a-zA-Z]+)?$", message = "Name must contain only alphabets and at most one space")
    private String name;
    
    @Email(message = "Email must be valid")
    private String email;
    
    @Pattern(
    	    regexp = "^(\\S+\\s+){0,29}\\S*$",
    	    message = "Bio must not exceed 30 words"
    	)
    private String bio;
    
    private String location;
    
    @URL(message = "Website must be a valid URL")
    private String website;
    
    private String skills;
    
    private String professionalName;
    private String channelName;
}

File: esports-backend-dotnet/Upkart1/DTO/UserProfileUpdateDTO.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Upkart1.DTO
{
    public class UserProfileUpdateDTO
    {
        [RegularExpression(@"^[a-zA-Z]+(\s[a-zA-Z]+)?$", ErrorMessage = "Name must contain only alphabets and at most one space")]
        [JsonPropertyName("name")]
        public string? Name { get; set; }

        [EmailAddress(ErrorMessage = "Email must be valid")]
        [JsonPropertyName("email")]
        public string? Email { get; set; }

        [RegularExpression(@"^(\S+\s+){0,29}\S*$", ErrorMessage = "Bio must not exceed 30 words")]
        [JsonPropertyName("bio")]
        public string? Bio { get; set; }

        [JsonPropertyName("location")]
        public string? Location { get; set; }

        [JsonPropertyName("website")]
        public string? Website { get; set; }

        [JsonPropertyName("skills")]
        public string? Skills { get; set; }
        
        [JsonPropertyName("professionalName")]
        public string? ProfessionalName { get; set; }
        
        [JsonPropertyName("channelName")]
        public string? ChannelName { get; set; }
    }
}